File: Applied Course/4.Problem Solving/1.Problems on Array/7.Create Target Array in the Given Order.py
# https://leetcode.com/problems/create-target-array-in-the-given-order/
class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        length=len(nums)
        target=[]
        for i in range(length):
            # if my index value grater than the length of the insert at the end
            if index[i]>=length:
                target.append(nums[i])
            else:
                target.insert(index[i],nums[i])
        return target

File: Applied Course/4.Problem Solving/1.Problems on Array/7.Create Target Array in the Given Order.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> v;
        for(int i=0;i<nums.size();i++)
            v.insert(v.begin()+index[i],nums[i]);
        return v;   
    }
};
int main()
{
    // https://leetcode.com/problems/create-target-array-in-the-given-order/
    return 0;
}